### PR TITLE
add zfs_setunlink() to check access rights before calling vnode_setunlink()

### PR DESCRIPTION
**Pull request by jheuking**
follow up #48

### CODE REVIEW

File: ZFSin/zfs/module/zfs/zfs_vnops_windows.c [hunk 1]
```diff
@@ -341,6 +341,88 @@ int zfs_find_dvp_vp(zfsvfs_t *zfsvfs, char *filename, int finalpartmaynotexist,
...
+	/*
+		if dvp == null, find it
+	*/
+	if (dvp == NULL) {
+		uint64_t parent;
+		if (sa_lookup(zp->z_sa_hdl, SA_ZPL_PARENT(zfsvfs),
+			&parent, sizeof(parent)) != 0) {
+			goto err;
+		}
+		if (zfs_zget(zfsvfs, parent, &dzp)) {
```
Comment (lundman) [inline ZFSin/zfs/module/zfs/zfs_vnops_windows.c]: `zget` will automatically do a HOLD on the dvp...

File: ZFSin/zfs/module/zfs/zfs_vnops_windows.c [hunk 2]
```diff
@@ -341,6 +341,88 @@ int zfs_find_dvp_vp(zfsvfs_t *zfsvfs, char *filename, int finalpartmaynotexist,
...
+	else {
+		dzp = VTOZ(dvp);
+	}
+
+	if (!dvp) {
+		Status = STATUS_NO_SUCH_FILE;
+		goto err;
+	}
+		
+	VN_HOLD(dvp);
```
Comment (lundman) [inline ZFSin/zfs/module/zfs/zfs_vnops_windows.c]: So this HOLD should be moved into the else that doesnt call `zget` at line 388.

File: ZFSin/zfs/module/zfs/zfs_vnops_windows.c [hunk 3]
```diff
@@ -341,6 +341,88 @@ int zfs_find_dvp_vp(zfsvfs_t *zfsvfs, char *filename, int finalpartmaynotexist,
...
+
+	int error = zfs_zaccess_delete(dzp, zp, 0);
+
+	if (error == 0) {
+		vnode_setunlink(vp);
+		Status = STATUS_SUCCESS;
+	}
+	else
+	{
```
Comment (lundman) [inline ZFSin/zfs/module/zfs/zfs_vnops_windows.c]: I guess we should decide on the coding style too, either stay with ZFS like cstyle.pl, or allow VC++ style too

**Comment by jheuking** [comment]
reworked in #50

File: ZFSin/zfs/module/zfs/zfs_vnops_windows.c (hunk 3 above)
Comment (jheuking) [reply ZFSin/zfs/module/zfs/zfs_vnops_windows.c]: I think staying with cstyle.pl would be ok. But in that case the script and configuration for checking the code style should at least be included in this repository...and maybe even better as a pre-commit hook?

Comment (lundman) [reply ZFSin/zfs/module/zfs/zfs_vnops_windows.c]: Yes, cstyle.pl should be included, and is in osx. Just not sure how to run perl in windows yet :)

File: ZFSin/zfs/module/zfs/zfs_vnops_windows.c (hunk 2 above)
Comment (jheuking) [reply ZFSin/zfs/module/zfs/zfs_vnops_windows.c]: oh ok? I wondered about that, but i found the acquiring of holds in zfs_zget was indeed somewhat confusing...

**Pull request closed without merging**